src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -202,6 +202,58 @@ public static Version GetInstalledAsmVersion2(IEnumerable<Version> versions, ref
             return null;
         }
 
+        /// <summary>
+        /// Get the corresponding libG preloader location for the target ASM loading version.
+        /// If there is exact match preloader version to the target ASM version, use it, 
+        /// otherwise use the closest below.
+        /// </summary>
+        /// <param name="version">The target loading version of ASM.</param>
+        /// <param name="rootFolder">Full path of the directory that contains 
+        /// LibG_major_minor_build folder. In a 
+        /// typical setup this would be the same directory that contains Dynamo 
+        /// core modules. This must represent a valid directory.</param>
+        /// <returns></returns>
+        public static string GetLibGPreloaderLocation(Version version, string rootFolder)
+        {
+            if (version == null)
+            {
+                version = new Version(0, 0, 0);
+            }
+
+            var libGFolderName = string.Format("libg_{0}_{1}_{2}", version.Major, version.Minor, version.Build);
+            var dir = new DirectoryInfo(Path.Combine(rootFolder, libGFolderName));
+            if (dir.Exists)
+            {
+                return dir.FullName;
+            }
+            else
+            {
+                // This usually means libG preloader version is behind the target version
+                var rootDir = new DirectoryInfo(rootFolder);
+
+                // Use regex to get all the libG versions supported
+                var libgFolders = rootDir.EnumerateDirectories("libg_*", SearchOption.TopDirectoryOnly);
+                var regExp = new Regex(@"^libg_(\d\d\d)_(\d)_(\d)$", RegexOptions.IgnoreCase);
+                var preloaderVersions = new List<Version>();
+                foreach (var folder in libgFolders)
+                {
+                    var match = regExp.Match(folder.Name);
+                    if (match.Groups.Count == 4 && Convert.ToInt32(match.Groups[1].Value) <= version.Major)
+                    {
+                        preloaderVersions.Add(new Version(
+                                Convert.ToInt32(match.Groups[1].Value),
+                                Convert.ToInt32(match.Groups[2].Value),
+                                Convert.ToInt32(match.Groups[3].Value)));
+                    }
+                }
+                preloaderVersions.Sort();
+                preloaderVersions.Reverse();
+                // Pick the closest preloader version below or use the default value when no libG folder found
+                var preloaderVersion = preloaderVersions.FirstOrDefault() == null ? version : preloaderVersions.First();
+                return Path.Combine(rootFolder, string.Format("libg_{0}_{1}_{2}", preloaderVersion.Major, preloaderVersion.Minor, preloaderVersion.Build));
+            }
+        }
```

Comment:
@QilongTang so I'm trying to understand this ... you're **not** taking into account the case here where an ASM installed version like ASM 224.5 is available on a user's machine and it still picks `LibG_224_4_0` folder to load LibG from. You're only looking at the latest ASM version (225 currently) and making sure that if the user has a version of ASM 225.X installed where X > Y, it still loads libg from `LibG_ASM225_Y_Z`? Am I right?

Reply:
@QilongTang I'm talking about 2 individual cases: one, where the user has ASM 224.X, where X > 4 and the other where the user has ASM 225.X, where X > 0 (what we have currently).

Reply:
@aparajit-pratap As I said, it depends on if the user has both Revit 2020 and Revit 2019.X installed locally and choose to just launch DynamoSandbox without Revit. If that is the case, Sandbox prefer ASM 225.X, if user only have one version of ASM locally, either 225.X or 224.X, Sandbox only work with that version. Does that answer your question?